src/pages/accounts/AdminAccountsPage.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { Fade, Stack, duration, Container } from "@mui/material";
+import { LayoutContainer } from "../../shared/LayoutContainer";
+import { useAppDispatch, useAppSelector } from "../../app/common/hooks";
+import {
+  requestAccounts,
+  unsetSelectedAccount,
+} from "./AdminAccountsPage.slice";
+import {
+  AccountsTable,
+  AccountDialog,
+  NewAccountDialog,
+  DashboardProgress,
+  AccountSnackbar,
+} from "./components";
+
+export function AdminAccountsPage() {
+  const dispatch = useAppDispatch();
+  const dashboardState = useAppSelector((state) => state.accountManager);
+
+  // whether there is a selected account from the table
+  const selectingAccount = dashboardState.selectedAccount ? true : false;
+
+  // getting accounts or not
+  const loading = dashboardState.loading;
+
+  // whether we want start transitioning
+  const [displayTable, setDisplayTable] = React.useState<boolean>(false);
+
+  // whether we want to add a new account (dialog)
+  const [newUserDialog, setNewUserDialog] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    dispatch(requestAccounts());
+  }, [dispatch]);
+
+  // for a smooth transition
+  React.useEffect(() => {
```

Comment:
Why use 2 useEffects instead of one?

Reply:
They have different effects. One is tied to dispatch and the other is tied to the loading status.

Reply:
> Why use 2 useEffects instead of one? 

> They have different effects. One is tied to dispatch and the other is tied to the loading status.

Yes, that is what I thought, if you think one way is better than the other, let me know where I can read more about it, thank you. Otherwise, I think I will leave it.